### PR TITLE
CLOUDP-179041: unify go version for ci/cd

### PR DESCRIPTION
**Pull request by wtrocki**
## Description

CI/CD process is using some old versions of Golang. 
Unifying everything to run on 1.20.

### CODE REVIEW

File: .github/workflows/pr.yml [hunk 1]
```diff
@@ -15,7 +15,7 @@ jobs:
       - name: Install Go
         uses: actions/setup-go@v4
         with:
-          go-version: 1.19
+          go-version: '1.20'
```
Comment (wtrocki) [inline .github/workflows/pr.yml:18]: We should be using 1.20 for GA everywhere. 1.19 support is almost guaranteed when we CI/CD against 1.20

**Comment by fmenezes** [comment]
why does our `go.mod` say 1.18 and our CI uses 1.20? is it intended?

File: .github/workflows/pr.yml [hunk 2]
```diff
@@ -15,19 +15,18 @@ jobs:
...
-          version: v1.50.1
+          version: v1.52.2
 
   tests-on-unix:
     needs: golangci-lint # run after golangci-lint action to not produce duplicated errors
     runs-on: ubuntu-latest
     strategy:
       matrix:
         golang:
-          - "1.19"
```
Comment (wtrocki) [inline .github/workflows/pr.yml:30]: The only controversial change here. I do not see benefit of us verifying under 2 versions of golang. 
We will likely to introduce another version soon so for GA we should have CI/CD process on single version for simplicty. 

Linting, formatting, introspections are all version based.

Comment (fmenezes) [reply .github/workflows/pr.yml:30]: should we remove the matrix then?

Comment (wtrocki) [reply .github/workflows/pr.yml:30]: Good point. We can always restore it for 1.21

File: .github/workflows/release-check.yml [hunk 3]
```diff
@@ -11,7 +10,7 @@ jobs:
     strategy:
       matrix:
         golang:
-          - 1.18
+          - '1.20'
```
Comment (fmenezes) [inline .github/workflows/release-check.yml:13]: you can remove the step `actions/cache@v3` your `actions/setup-go@v4` already caches it

Comment (fmenezes) [reply .github/workflows/release-check.yml:13]: same for `.github/workflows/pr.yml`

File: .github/workflows/pr.yml (hunk 2 above)
Comment (fmenezes) [reply .github/workflows/pr.yml:30]: if we intend to restore it, there is no need to remove

File: .github/workflows/release-check.yml (hunk 3 above)
Comment (wtrocki) [reply .github/workflows/release-check.yml]: Awesome catch! it will make entire pipeline much cleaner.

**Pull request merged**